syntax/nodes.go

```diff
@@ -168,6 +170,7 @@ type Comment struct {
 
 func (c *Comment) Pos() Pos { return c.Hash }
 func (c *Comment) End() Pos { return posAddCol(c.Hash, 1+len(c.Text)) }
+func (c *Comment) IsShebang() bool { return fileutil.Shebang([]byte("#"+c.Text)) != "" && c.Hash.Col() == 1 && c.Hash.Line() == 1 }
```

Comment:
I would prefer to not add new API here; `fileutil.Shebang` is always available to end users. You can move the three conditions into printer.go.